packages/api/src/external/carequality/command/test-and-update-patient-demographics.ts

```diff
@@ -0,0 +1,70 @@
+import { Patient } from "@metriport/core/domain/patient";
+import { Address } from "@metriport/core/domain/address";
+import { getPatientOrFail } from "../../../command/medical/patient/get-patient";
+import { PatientModel } from "../../../models/medical/patient";
+import { executeOnDBTx } from "../../../models/transaction-wrapper";
+import { InboundPatientResource } from "@metriport/ihe-gateway-sdk";
+
+type PatientResourceAddres = InboundPatientResource["address"][number];
+
+export async function testAndUpdatePatientDemographics({
+  patient,
+  patientResource,
+}: {
+  patient: Pick<Patient, "id" | "cxId">;
+  patientResource: InboundPatientResource;
+}): Promise<boolean> {
+  const patientFilter = {
+    id: patient.id,
+    cxId: patient.cxId,
+  };
+
+  return await executeOnDBTx(PatientModel.prototype, async transaction => {
+    const existingPatient = await getPatientOrFail({
+      ...patientFilter,
+      lock: true,
+      transaction,
+    });
+
+    const newAddresses: PatientResourceAddres[] = patientResource.address.flatMap(
+      (newAddress: PatientResourceAddres) => {
+        const isNew: boolean = existingPatient.data.address.every(existingAddress =>
+          checkNonMatchingAddress(newAddress, existingAddress)
+        );
+        if (isNew) return newAddress;
+        return [];
+      }
+    );
+
+    if (!newAddresses) {
+      return false;
+    }
+
+    const newAddress = [...existingPatient.data.address, ...newAddresses];
+
+    const updatedPatient = {
+      ...existingPatient.dataValues,
+      address: newAddress,
+    };
+
+    await PatientModel.update(updatedPatient, {
+      where: patientFilter,
+      transaction,
+    });
+
+    return true;
+  });
+}
+
+export function checkNonMatchingAddress(
+  address1: PatientResourceAddres,
+  address2: Address
+): boolean {
+  return (
+    (address1.line && address1.line.length > 0 && address1.line[0] !== address2.addressLine1) ||
```

Comment:
if there are potentially multiple address lines in a single inbound address, arent we potentially losing them by 0 indexing here? Maybe IRL we never actually have multiple lines inside, or maybe they aare always part of the ["788 Columbus Ave", "Apt 8G"] multi part address line schema, but either way, would be good to either understand the data we have to justify this or to either .join the whole array or to iterate through address lines as well